llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -79,6 +79,10 @@ static cl::opt<LoopVectorizeHints::ScalableForceKind>
                 "Scalable vectorization is available and favored when the "
                 "cost is inconclusive.")));
 
+static cl::opt<bool>
+    EnableCSA("enable-csa-vectorization", cl::init(false), cl::Hidden,
+              cl::desc("Control whether CSA loop vectorization is enabled"));
```

Comment:
Why guard it under a flag? Seems useful in the general case.

Reply:
This is a new feature, and I think it should be able to be toggled by a switch at least until the feature becomes more mature. This patch is disabling CSA by default (except for the sifive7 which was measured to see benefit). We can consider enabling it across the board in a future patch.

https://github.com/llvm/llvm-project/pull/83467 takes the same approach.